sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -363,6 +386,14 @@ static RunQueryRequest makeRequest(Query query, @Nullable String namespace) {
      */
     public DatastoreV1.Read withProjectId(String projectId) {
       checkNotNull(projectId, "projectId");
+      return toBuilder().setProjectId(StaticValueProvider.of(projectId)).build();
+    }
+
+    /**
+     * Same as {@link Read#withProjectId(String)} but with a {@link ValueProvider}.
+     */
+    public DatastoreV1.Read withProjectId(ValueProvider<String> projectId) {
+      checkNotNull(projectId, "projectId");
```

Comment:
We're now following a different style for the error messages in these checks. See e.g. JdbcIO.